src/wallet/rpcwallet.cpp

```diff
@@ -2513,24 +2513,25 @@ static UniValue loadwallet(const JSONRPCRequest& request)
             }.Check(request);
 
     WalletContext& context = EnsureWalletContext(request.context);
-    WalletLocation location(request.params[0].get_str());
+    const std::string name(request.params[0].get_str());
+    fs::path path(fs::absolute(name, GetWalletDir()));
 
-    if (!location.Exists()) {
-        throw JSONRPCError(RPC_WALLET_NOT_FOUND, "Wallet " + location.GetName() + " not found.");
-    } else if (fs::is_directory(location.GetPath())) {
+    if (fs::symlink_status(path).type() == fs::file_not_found) {
```

Comment:
in commit 83e4ead9917735b2ba92c8a35fe37504206ffe32

please don't change behavior in refactoring commits.

Previously:

```
test_framework.authproxy.JSONRPCException: Wallet  not found. (-18)
```

Now:

```
test_framework.authproxy.JSONRPCException: Directory  does not contain a wallet.dat file. (-18)

Reply:
I sort of liked that the functionality was wrapped in one function. This avoids issues exactly like this